addons/misra.py

```diff
@@ -256,15 +256,58 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand2]
 
 
-def hasFloatComparison(expr):
-    if not expr:
-        return False
-    if expr.isLogicalOp:
-        return hasFloatComparison(expr.astOperand1) or hasFloatComparison(expr.astOperand2)
-    if expr.isComparisonOp:
-        # TODO: Use ValueType
-        return cppcheckdata.astIsFloat(expr.astOperand1) or cppcheckdata.astIsFloat(expr.astOperand2)
-    return False
+def findCounterTokens(cond):
+    if cond.str in ['&&', '||']:
+        c = findCounterTokens(cond.astOperand1)
+        c.extend(findCounterTokens(cond.astOperand2))
+        return c
+    ret = []
+    if ((cond.isArithmeticalOp and cond.astOperand1 and cond.astOperand2) or
+            (cond.isComparisonOp and cond.astOperand1 and cond.astOperand2)):
+        if cond.astOperand1.isName:
+            ret.append(cond.astOperand1)
+        if cond.astOperand2.isName:
+            ret.append(cond.astOperand2)
+        if cond.astOperand1.isOp:
+            ret.extend(findCounterTokens(cond.astOperand1))
+        if cond.astOperand2.isOp:
+            ret.extend(findCounterTokens(cond.astOperand2))
+    return ret
+
+
+def isFloatCounterInWhileLoop(whileToken):
+    if not simpleMatch(whileToken, 'while ('):
+        return None
+    lpar = whileToken.next
+    if not lpar.astOperand2.astOperand1:
+        return None
+    rpar = lpar.link
+    counter = findCounterTokens(lpar.astOperand2)
+    if len(counter) == 0:
```

Comment:
It seems safe to remove this code:

    if len(counter) == 0:
        return None